### ISSUE TITLE
A bootstrap floor of exactly zero still licenses every delta outside the ladder

### ISSUE THREAD

**Issue by mmmfrieddough** [mmmfrieddough/anthro-chess#304]
## What was observed

#190 established a rule for the ladder: a quantity every resample reproduced exactly gets no floor, because a bootstrap that could not move a number has not measured an absence of noise, and a floor of zero clears every subsequent delta. The genuine zero — a reading that replays — is stated and labelled as stated, per decision 0032.

Two other call sites run the same estimator shape and neither refuses a dispersion of zero:

- `anthro_chess.evaluation.curves`, which bootstraps generated games and calls `floor_from_dispersion`.
- `anthro_chess.evaluation.noise.bootstrap_floors`, which omits a metric only when fewer than two finite replicates exist.

A curve reading whose model side reproduces its distance exactly therefore still emits a floor of `0.0`, which is the `0.0000` ambiguity #175 raises arriving through a different route.

## Why this is not simply the same fix

Decision 0032 *deliberately* states a floor of zero for a curve comparison whose model side cannot vary, and argues that reporting no floor there understates what is known. So the two zeros already mean different things in that family, and a blanket refusal in the shared arithmetic would remove the one that is correct.

The question is whether an *estimated* zero is distinguishable from a *stated* one at each call site, and what each should report. In the ladder they are, because the two come out of different branches carrying different methods.

## Scope

Decide, per family, whether a dispersion of zero from a resample is a floor or a refusal, and whether the rule belongs beside `bounded_floor`/`floor_from_dispersion` rather than at each call site. Then implement it.

Where a floor is refused, the output has to distinguish it from a floor nobody has produced yet, which is the same requirement #190 and #175 both carry.

## References

- #190 — where this was found; `docs/decisions/0034-qualifying-a-rating-ladder-reading.md`
- #175 — a floor rendering as exactly `0.0000`
- `docs/decisions/0032-a-replayed-reading-has-no-evaluation-noise.md`
- `docs/decisions/0026-conservative-dispersion-bounds.md`

**Comment by mmmfrieddough** [comment]
A measurement from #173 that bears on this, one step short of the zero.

The stratified bootstrap does not only collapse at one unit per stratum — it
understates by a knowable factor at two, and that is the setting the reduced
sweep ships. The plug-in variance inside a stratum of `n` is `((n-1)/n)s²`, so
the resample mean's variance is `((n-1)/n²)s²`. Between two puzzles per rating
and ten that predicts a spread ratio of `sqrt(0.25/0.09) = 1.67`, where the
sample-size argument alone predicts `sqrt(5) = 2.24`.

Measured on `issue-203-treatment` step 8000, `anthro eval puzzles`, the new
within-reading refit spread of the fitted puzzle rating:

| | 4,000 puzzles (2 per rating) | 20,000 puzzles (10 per rating) | ratio |
| --- | ---: | ---: | ---: |
| greedy | ±35.4 | ±20.9 | 1.69 |
| sampled | ±19.2 | ±11.0 | 1.75 |

Against 1.67 predicted and 2.24 from `1/√n` alone. So the reduced sweep's
stratified spreads are roughly 25% narrower than the population they describe,
in the direction a floor exists to prevent.

This is the same mechanism as the zero here, not a separate one: the retained
paired contributions stratify by exact rating identically, so the puzzle floors
a reduced sweep records carry it too. The dispersion bound does not compensate,
because its degrees of freedom count the units (`len(unit_ids) - 1`) rather than
anything about the strata — with 4,000 units the bound widens by about 1.9%
whatever the stratum size is.

Recorded here rather than filed separately, since whatever fixes a zero at one
unit per stratum should decide what happens at two.

Refs #173, #168.

**Comment by mmmfrieddough** [comment]
Follow-up: #354 landed `0039-stratifying-the-ladder-redraw-costs-more-than-it-removes.md`
between my comment above and #173 being ready, and it reaches the same
conclusion from the ladder side — including the correction, which my comment
stopped short of naming.

Two things that changes here.

**The measurement replicates the record, on a second family.** #173's new
within-reading refit takes the rescaled draw (`n-1` of each stratum, counts
scaled back up). Same checkpoint, same two set sizes, fitted puzzle rating:

| draw | 4,000 → 20,000 spread ratio |
| --- | ---: |
| plug-in | 1.69x (greedy), 1.75x (sampled) |
| rescaled | 2.29x (greedy), 2.26x (sampled) |

against `√5 = 2.24`. So the correction restores `1/√n` exactly where the record
says it should, at a two-unit stratum rather than the ladder's three.

**What is left here is the retained paired contributions.** `draw_multiplicity`
in `anthro_chess.evaluation.results.paired` now takes a `rescale` flag and
`PairedFloorIndex` deliberately does not pass it, because turning it on moves
floors already recorded. At the reduced sweep's `puzzles_per_rating=2` those
floors are the factor above too narrow, which is this issue's failure one step
short of its limit. The mechanism is now implemented and tested
(`test_a_rescaled_stratified_draw_removes_the_plug_in_understatement`), so the
remaining work is deciding what to do about the readings already in the store
rather than writing the estimator.

**Comment by mmmfrieddough** [comment]
Part of this is fixed in #361, and the part that is not turned out to be the expensive part.

**Fixed there:** the paired estimator now takes the rescaled draw, and a
stratum of one withholds the paired floor with a stated reason rather than
returning the confident zero this issue names. Nothing recorded changes — a
paired floor is derived at report time and never stored.

**Still open here, and worth knowing before someone picks it up.** I implemented
the refusal at the two call sites this issue actually names —
`evaluation.noise.bootstrap_floors` and the curve family — and reverted it. It
turns nine readings across `test_rollout`, `test_termination_benchmark`,
`test_checkpoint` and `test_results_noise` into unqualified ones. Every failure
is a fixture too thin to show spread (`model_games=2`, or a rate identical in
every fixture game) rather than a disagreement about the rule, so the work is
not the refusal — it is deciding what those fixtures should be, and whether a
fixture-scale reading legitimately has no floor.

One of them is load-bearing rather than incidental:
`test_bootstrap_resamples_games_rather_than_positions` asserts
`identical[0].floor == 0.0` for identical games, which is this issue's claim
written down as a specification. And
`test_a_paired_floor_preserves_the_benchmark_strata` documents a paired floor of
exactly `0.0` as correct by design, because each stratum's delta is constant.
So the per-family decision this issue asks for has to say which of those two
zeros survives.

**Comment by mmmfrieddough** [comment]
Half of this is fixed in #361, and the remaining half turned out to be one specific, nameable thing rather than a per-family judgement call.

**Fixed there**

- `evaluation.noise.bootstrap_floors` omits a metric whose bootstrap dispersion
  is exactly zero, as it already omitted one whose dispersion could not be
  estimated. Two fixtures asserted the old shape and now state the rule —
  including `test_bootstrap_resamples_games_rather_than_positions`, which
  asserted `identical[0].floor == 0.0`, this issue's complaint written down as a
  specification.
- The paired estimator withholds its floor at a stratum of one, with a stated
  reason, instead of returning the confident zero a redraw of a single unit
  produces.

**Answering the "where does the rule belong" question:** at the call site. The
predicate is one comparison, what to *do* about a zero differs by caller, and
putting it in `bounded_floor`/`floor_from_dispersion` would also catch `0032`'s
stated zero, which is the one that is correct.

**What is left, and why it is not two lines**

Only the curve family. The change itself *is* two lines, and in production it
would never fire — a curve over hundreds of varied games always moves. It fires
only against the fixtures, and the cause is specific:

- `TrajectoryRunner.predict` seeds on `len(context.plies)` and the target rating
  alone, so every game in a cell is the same game;
- `maximum_generated_plies = 6` truncates all of them at the same length.

So the model side cannot vary, and no redraw can move a distance. I measured
this rather than assuming it: adding a stub whose logits differ per call, with
`games_per_position=4`, left `generated_play.game_length_conditional_distance`
at exactly `1.5999999999999996` — unchanged, floor still absent. A varying stub
is not enough; the ply limit has to let games end at different lengths too, and
that moves the counts a dozen unrelated assertions in `test_rollout.py` read
(`test_a_finished_game_counts_toward_the_rates_computed_over_results`,
`test_the_ply_limit_is_reported_as_unfinished_rather_than_as_a_result`,
`test_a_cell_reports_book_depth_in_three_parts`, and others — raising
`games_per_position` alone broke five).

Six tests read a curve floor: five in `test_rollout.py` and
`test_the_mix_reports_each_arm_beside_its_own_qualifiers` in
`test_termination_benchmark.py`. Rewriting their assertions to expect no floor
would be weakening them to pass — `test_a_distance_carries_the_floor_it_has_to_clear`
would then assert the opposite of its name. So this needs a fixture that
generates genuinely varied play, which is that family's architecture rather than
a change to the rule.

Recorded in `docs/decisions/0041-the-puzzle-response-is-qualified-within-its-reading.md`.

**Comment by mmmfrieddough** [comment]
Settled and closing on #361, which decides both halves of the per-family question this issue asks.

**A bootstrap over per-game contributions refuses an estimated zero.**
`evaluation.noise.bootstrap_floors` omits a metric whose dispersion is exactly
zero, as it already omitted one whose dispersion could not be estimated. The
paired estimator withholds at a stratum of one, with a stated reason, instead of
returning the confident zero a redraw of a single unit gives. Two fixtures
asserted the old shape and now state the rule — including
`test_bootstrap_resamples_games_rather_than_positions`, which asserted
`identical[0].floor == 0.0`, this issue's complaint written down as a spec.

**Where the rule belongs:** at the call site. The predicate is one comparison,
what to *do* about a zero differs by caller, and putting it in `bounded_floor`
would also catch `0032`'s stated zero, which is correct.

**A curve over generated play keeps its estimated zero.** There the games *are*
the draw (`0032`), so a distance no redraw moves describes the play this reading
generated rather than a sample that came out flat — and `CurveComparison`
already carries `model_variation` beside the distance, so a reader sees the
model side did not move without inferring it from the floor.

That is a decision rather than a deferral, and it was measured before being
made. Refusing there is two lines and never fires in production. At fixture
scale it removes floors from whichever quantities the fixture cannot exercise,
and which ones those are shifts with the configuration:

| fixture | quantities left without a floor |
| --- | --- |
| one game per cell (as shipped) | all ten |
| varying stub, resignation on, 4 games/cell | four — repetition, cycle, book depth, move diversity |
| the same plus a third seed | five, and the depth sweep loses its floor at the plies games no longer reach |

Six tests across `test_rollout.py` and `test_termination_benchmark.py` assert
that *every* quantity carries a floor. Narrowing them to whichever subset a
given fixture happens to move would leave
`test_a_distance_carries_the_floor_it_has_to_clear` asserting less than its name
claims — a worse outcome than a zero the reading already qualifies.

Recorded in `docs/decisions/0041-the-puzzle-response-is-qualified-within-its-reading.md`.

**Comment by mmmfrieddough** [comment]
## Three more call sites, and the one that writes zeros into the store

An audit of the floor system found this rule broken in **five** producers, not two. `ladder.py:1497` is the only one that refuses a zero dispersion. Adding to what this issue already records:

- `results/noise.py:replicate_floors` — the estimator behind `anthro eval noise characterize --kind evaluation|training`.
- `results/noise.py:process_replicate_floors` — the execution floor.
- `results/paired.py:306-318` — the paired estimator.

Every one of them reaches `dispersion_bound`, and `dispersion_bound(0.0, degrees_of_freedom=n)` returns `0.0` without complaint. That is the single chokepoint this issue asks about: a guard there covers all five.

## The CLI path is worse than the bootstrap path

`replicate_floors` writes a *stored characterization*, so one command poisons many metrics at once and the result outlives the session. Reproduced on a copy of the committed store:

```
anthro eval noise characterize --kind evaluation \
  --checkpoint training-blitz-30k-v4-step-00000100 \
  --checkpoint training-blitz-30k-v4-step-00000100 --source "probe"

→ "Characterized evaluation noise over 2 replicate(s) for 61 metric(s)."
→ all 61 floors 0.0, no warning
```

`_collect_replicates` (`cli.py:3174`) takes `latest_measurement` per label, so a repeated label yields the same reading twice and a dispersion of exactly zero. Because the most recent characterization of a series wins (`results/noise.py:357`), those zeros then override the legitimate bootstrap floors.

Report verdicts on the committed pair, before and after:

| | cleared | within | unknown |
| --- | --- | --- | --- |
| committed store | 44 | 10 | 5 |
| after the command above | 51 | 10 | 0 |

Five metrics that honestly reported `unknown` became `cleared`. `_noise_verdict` (`reporting.py:1652`) tests `abs(delta) <= floor`, so a floor of `0.0` returns `CLEARED` for any nonzero delta while a *missing* floor returns `UNKNOWN` — the two zeros take opposite branches.

## Already present in the committed store

`results/floors/20260801T073755Z-data-sampling-74b973ee79d1b400.json` carries four floors of exactly `0.0`: `adjudicated.mate_available_selected_rate`, `adjudicated.only_move_best_rank`, `adjudicated.only_move_selected_rate`, `adjudicated.stalemate_available_selected_rate`.

## The fix has a test to change

`tests/evaluation/test_results_noise.py:180` asserts `identical[0].floor == 0.0`, so it pins the current behaviour and has to move with the fix.

## Suggested shape

Reject a zero dispersion in `dispersion_bound` and let each caller convert that into its own refusal, which keeps decision 0032's *stated* zero (a distinct construction, not an estimate) intact. Separately, `_collect_replicates` should reject duplicate labels outright — that is a caller error rather than a measurement.
